src/server/node_manager.c

```diff
@@ -5128,6 +5129,7 @@ is_request(int stream, int version)
 			}
 
 			free_attrlist(&reported_hooks);
+			psvrmom->msr_is_hook_checksums_done = 1;
```

Comment:
Have you consider the situation, when no sync is required (i.e Hooks are  already synced).